### DISCUSSION TITLE
Approach to construct a race track

### DISCUSSION THREAD

**Discussion by jtodevs** [juanmanzanero/fastest-lap#15] [Q&A] [answered]
Hi,
Thanks for this tool, I had this idea in my backlog for a while, but never found time to develop it!

I constructed a race track with the hints you provide in the intro (kml file through Google Earth, and processing with the Python tool included in this repo to produce the xml file). Every thing worked fine, however, after optimising, I notived the track is not followed in the correct direction (which should be clockwise overall). I tried to construct the polyline of the track going clockwise and counter clockwise, with no effect. I guess there is a kml convention.
Eventually, I patched a bit of the code, and it works now as expected.

What is actually the correct approach ?
thanks!

**Comment by juanmanzanero** [comment, marked as answer]
I am surprised this happens. The direction should follow that of the kml files. Can I have the KML files and the python script to debug it?

**Reply by jtodevs** [reply]
Here is an archive. I used the web version of Google Earth.
[karting_track.tar.gz](https://github.com/juanmanzanero/fastest-lap/files/8938150/karting_track.tar.gz)

In any case, it could be a nice feature to "reverse" the track direction when importing a kml.

**Reply by jtodevs** [reply]
Hi,
I think I found out my problem. I was closing the path under Google Earth and then extracting points from KML to produce a new KML with the track. Doing so, Google Earth actually always produces a anti clockwise path.

**Reply by juanmanzanero** [reply]
Good! Feel free to upload the track files, python scripts, and track XML files to the repository by creating a pull request.
